token_to_id being list, type annotations

allenoxli opened the pull request.

allenoxli committed:
token_to_id being list, type annotations

ProFatXuanAll commented on example_generate.py:
```diff
@@ -2,27 +2,39 @@
-experiment_no = 1
+
+# experiment_no = 1
+experiment_no = 5
```
remove unnecessary comment and change it to `experiment_no = 1`

ProFatXuanAll commented on example_generate.py:
```diff
@@ -2,27 +2,39 @@
+
+# model = lmp.model.GRUModel(config=config,
+#                            tokenizer=tokenizer)
+model = lmp.model.LSTMModel(config=config,
```
remove unnecessary comment

ProFatXuanAll commented on example_generate.py:
```diff
@@ -2,27 +2,39 @@
+
+
+
+BOS = sys.argv[1] if len(sys.argv) > 1 and sys.argv[1] != ''  else '今天'
```
remove unnecessary code

ProFatXuanAll commented on example_train.py:
```diff
@@ -11,14 +11,15 @@
 # Hyperparameters setup
 ##############################################
-experiment_no = 1
+experiment_no = 6
```
change it to `experiment_no = 1`

ProFatXuanAll commented on example_train.py:
```diff
@@ -57,9 +58,13 @@
 tokenizer = lmp.tokenizer.CharTokenizer()
 
+# 讓使用者決定是否 uncase
+uncaseOrNot = True if len(sys.argv) > 1 and sys.argv[1].lower() == 'uncase'   else False
```
rewrite the script and use built-in `argparse`

ProFatXuanAll commented on example_train.py:
```diff
@@ -70,8 +75,11 @@
-model = lmp.model.GRUModel(config=config,
+model = lmp.model.LSTMModel(config=config,
                            tokenizer=tokenizer)
+# model = lmp.model.GRUModel(config=config,
```
remove unnecessary comment

ProFatXuanAll commented on lmp/config.py:
```diff
@@ -3,21 +3,31 @@
+                     dropout:float=0,
+                     embedding_dim:int=1,
+                     epoch:int=1,
+                     grad_clip_value:float=1,
```
change `grad_clip_value` to `max_norm`.

ProFatXuanAll commented on lmp/config.py:
```diff
@@ -41,7 +51,7 @@ def load_from_file(cls, file_path=None):
         return self
 
-    def save_to_file(self, file_path=None):
+    def save_to_file(self, file_path:str=None):
```
add space after colon

ProFatXuanAll commented on lmp/config.py:
```diff
@@ -3,21 +3,31 @@
-        self.num_rnn_layers = kwargs.pop('num_rnn_layers', 1)
-        self.num_linear_layers = kwargs.pop('num_linear_layers', 1)
-        self.seed = kwargs.pop('seed', 1)
+    def __init__(self, batch_size:int=1,
```
change style to
```py
def __init__(
        self,
        batch_size: int = 1,
        ...
):
```
add space after colon

ProFatXuanAll commented on lmp/dataset.py:
```diff
@@ -1,29 +1,39 @@
 
 class BaseDataset(torch.utils.data.Dataset):
-    def __init__(self, **kwargs):
+    def __init__(self, text_list: List[str],
```
change style to
```py
def __init__(
        self,
        text_list: List[str],
        ...
):
```
add space after colon

ProFatXuanAll commented on lmp/dataset.py:
```diff
@@ -1,29 +1,39 @@
+        self.text_list = text_list
+        self.config = config
+        self.tokenizer = tokenizer
+        self.uncase = uncase  # 是否把大小寫視為不同字, default: False
```
when value of variable is `bool` type, named the variable using prefix `is_`.
for example: `is_uncased`

ProFatXuanAll commented on lmp/tokenizer.py:
```diff
@@ -2,24 +2,193 @@
 
 class BaseTokenizer:
-    def __init__(self, **kwargs):
+    def __init__(self, pad_token: str = '[PAD]', pad_token_id: int = 0,
```
Name the classes `BaseTokenizerByList` and `BaseTokenizerByDict`

ProFatXuanAll commented on lmp/tokenizer.py:
```diff
@@ -2,24 +2,193 @@
+        self.unk_token_id = unk_token_id
+
+        self.token_to_id: List[str] = []  # 用 list 節省記憶體
+        self.id_to_token: Dict[int, str] = {}
```
inverse lookup should only use `self.token_to_id.index`

ProFatXuanAll commented on lmp/tokenizer.py:
```diff
@@ -2,24 +2,193 @@
+            ids = []
+            for token in tokens:
+                if token in self.token_to_id:
+                    # 取出該token在 list的index
```
prepend and append space around english words

allenoxli committed:
remove unnecessary code and comments, using argparse to determine uncase or not in example_train.py, remove id_to_token in BaseTokenizerByList, substitute clip_grad_norm_ for clip_grad_value_

ProFatXuanAll commented on example_generate.py:
```diff
@@ -11,29 +10,23 @@
 
-# experiment_no = 1
-experiment_no = 5
+experiment_no = 1
```
use `argparse` to get `experiment_no`, wrap whole generation script into a function.

ProFatXuanAll commented on example_train.py:
```diff
@@ -59,12 +59,15 @@
 # 讓使用者決定是否 uncase
-uncaseOrNot = True if len(sys.argv) > 1 and sys.argv[1].lower() == 'uncase'   else False
+parser = argparse.ArgumentParser()
+parser.add_argument("-u", "--uncase", help="regard capital letter and lowercase letter as same word",
```
Do not use short-hand argument (i.e. `"-u"`).
Change helper text into `"convert all upper case into lower case."`.
Indicate `--uncase` as required argument.

ProFatXuanAll replied:
rename `arg.uncase` into `arg.is_uncased` for consistency

ProFatXuanAll commented on example_train.py:
```diff
@@ -59,12 +59,15 @@
                                   text_list=df['title'],
                                   tokenizer=tokenizer,
-                                  uncase=uncaseOrNot)
+                                  is_uncased=args.uncase)
```
Put `uncase` into `config` object.
Parse all config option by `argparse`.

ProFatXuanAll commented on lmp/tokenizer.py:
```diff
@@ -135,8 +130,8 @@ def decode(self, all_ids: List[List[int]]) -> List[str]:
-    def build_dict(self, all_sentences: List[str], min_count: int = 0, uncase: bool = False):
-        if uncase:  # 如果uncase是True，就不分大小寫
+    def build_dict(self, all_sentences: List[str], min_count: int = 0, is_uncased: bool = False):
+        if is_uncased:  # 如果 is_uncased 是 True，就不分大小寫
```
Write docstring

allenoxli replied:
我需要把每個 function和class 都寫 docstring ?

allenoxli committed:
third commit, Parse all config option by argparse. Put is_uncased into config. Wrap example_generate as a function.

allenoxli committed:
forth commit, write docstring.

allenoxli committed:
fifth commit, write checkpoint

ProFatXuanAll commented on example_train.py:
```diff
@@ -11,26 +19,57 @@
+
+# Required arguments.
+parser.add_argument("--experiment_no", type=int, default=1, required=True, help="using which experiment_no data")
+parser.add_argument("--is_uncased",    type=boolean_string, default=False, required=True, help="convert all upper case into lower case.")
```
Using `--is_uncased` without passing value indicate `True`.
See https://stackoverflow.com/questions/15008758/parsing-boolean-values-with-argparse

ProFatXuanAll commented on example_train.py:
```diff
@@ -11,26 +19,57 @@
+parser.add_argument("--epoch",              type=int,   default=10,     help="Number of training epochs.")
+parser.add_argument("--max_norm",           type=float, default=1,      help="Max norm of gradient.Used when cliping gradient norm.")
+parser.add_argument("--hidden_dim",         type=int,   default=300,    help="Hidden dimension.")
+parser.add_argument("--learning_rate",      type=float, default=10e-4,  help="Optimizer's parameter `lr`.")
```
`10e-4 == 1e-3`, use the latter one.
This must be consider as bugs.

ProFatXuanAll replied:
Also change the default value to `5e-5`

ProFatXuanAll commented on example_train.py:
```diff
@@ -11,26 +19,57 @@
+    if s not in {'False', 'True'}:
+        raise ValueError('Not a valid boolean string')
+    return s == 'True'
+
```
wrap all `argparser` code into scope `if __name__ == '__main__':`

ProFatXuanAll commented on example_train.py:
```diff
@@ -11,26 +19,57 @@
-                               num_linear_layers=2,
-                               seed=7)
+experiment_no = args.experiment_no
+config = lmp.config.BaseConfig(batch_size=args.batch_size,
```
Pass argument in alphabetic order.
Style code into
```py
lmp.config.BaseConfig(
    batch_size=...,
    ...
)
```

ProFatXuanAll commented on example_train.py:
```diff
@@ -56,15 +95,13 @@
 # Construct tokenizer and perform tokenization.
 ##############################################
-tokenizer = lmp.tokenizer.CharTokenizer()
+tokenizer = lmp.tokenizer.CharTokenizerByList()
```
Tokenizer should be construct through `argparser`

allenoxli committed:
sixth commit, fix 10e-4 to 1e-4, --is-uncased use action='store_true' instead

ProFatXuanAll commented on example_train.py:
```diff
@@ -101,24 +136,36 @@
+    
     print(f'epoch {epoch}')
     total_loss = 0
+    iteration_times = 0
```
Change `iteration_times` into `step`

ProFatXuanAll commented on example_train.py:
```diff
@@ -77,8 +114,6 @@
 ##############################################
 model = lmp.model.LSTMModel(config=config,
```
`model` should be construct through `argparser`

ProFatXuanAll commented on example_generate.py:
```diff
@@ -11,30 +18,37 @@
+    model_path = f'{data_path}/{experiment_no}'
+
+    config = lmp.config.BaseConfig.load_from_file(f'{model_path}/config.pickle')
+    tokenizer = lmp.tokenizer.CharTokenizerByList.load_from_file(f'{model_path}/tokenizer.pickle')
```
Use `config` to construct `tokenizer` and `model`.

ProFatXuanAll commented on test.py:
```diff
@@ -0,0 +1,10 @@
+import argparse
```
Remove unused code.
If this file is intended to perform unit test, then use `unittest` or other framework to write unit test

allenoxli committed:
1.wrap all argparser code into __main__ scope
2.change the learning_rate default value to 5e-5
3.modify docstring
4.Pass argument in alphabetic order.
5.Change iteration_times into step
6.model should be construct through argparser
7.Tokenizer should be construct through argparser
8.Use config to construct tokenizer and model in example_generate.py.

allenoxli committed:
add util file in lmp, change load_from_file's way like this return cls(**) in config

allenoxli committed:
add tensorboar writter, save optimizer state to keep training model, modify dataset collate_fn

allenoxli committed:
update utility file

allenoxli committed:
update tokenizer and model file

allenoxli committed:
restructure files architecture.
add more comment for newbies.

allenoxli committed:
tokenizer class in config.py

allenoxli committed:
a little modify

ProFatXuanAll commented on lmp/config.py:
```diff
@@ -108,7 +108,7 @@ def __init__(
             optimizer_class: str = 'adam',
             seed: int = 1,
-            tokenizer_class: str = 'list'
+            tokenizer_class: str = 'char_dict''
```
syntax error

allenoxli committed:
YoYo
